5_data_munge/src/munge_wqp_files.R

```diff
@@ -93,19 +93,123 @@ munge_wqp_withoutdepths <- function(in_ind, min_value, max_value, max_daily_rang
+    filter(!is.na(StatisticalBaseCode)) %>%
+    group_by(MonitoringLocationIdentifier, ActivityStartDate) %>%
+    mutate(temperature_mean_daily = ifelse(grepl('mean', StatisticalBaseCode, ignore.case = TRUE), ResultMeasureValue, NA),
+           temperature_min_daily = ifelse(grepl('min', StatisticalBaseCode, ignore.case = TRUE), ResultMeasureValue, NA),
+           temperature_max_daily = ifelse(grepl('max', StatisticalBaseCode, ignore.case = TRUE), ResultMeasureValue, NA)) %>%
+    mutate(n_obs = NA) %>%
+    group_by(MonitoringLocationIdentifier, ActivityStartDate) %>%
+    summarize(across(c(temperature_mean_daily, temperature_min_daily, temperature_max_daily, n_obs) , ~ first(na.omit(.))))
+
+  # we don't know the number of observations here because stat codes were used
+
+
+  dat_daily <- ungroup(dat_reduced) %>%
+    filter(is.na(StatisticalBaseCode)) %>%
+    group_by(MonitoringLocationIdentifier, ActivityStartDate) %>%
+    summarize(n_obs = n(),
+              temperature_mean_daily = mean(ResultMeasureValue),
+              # we don't want to propagate min and max if there is only one value
+              temperature_min_daily = ifelse(n_obs>1, min(ResultMeasureValue), NA),
+              temperature_max_daily = ifelse(n_obs>1, max(ResultMeasureValue), NA),
               time = ifelse(n_obs == 1, `ActivityStartTime/Time`, NA)) %>%
-    filter(temperature_mean_daily > min_value & temperature_mean_daily < max_value,
-           temperature_min_daily > min_value & temperature_min_daily < max_value,
-           temperature_max_daily > min_value & temperature_max_daily < max_value)
+    bind_rows(dat_reduced_statcode) %>%
+    filter(temperature_mean_daily > min_value & temperature_mean_daily < max_value|is.na(temperature_mean_daily),
+           temperature_min_daily > min_value & temperature_min_daily < max_value|is.na(temperature_min_daily),
+           temperature_max_daily > min_value & temperature_max_daily < max_value|is.na(temperature_max_daily))
 
   # save
   data_file <- scipiper::as_data_file(out_ind)
   saveRDS(dat_daily, data_file)
   gd_put(out_ind)
 
 }
+
+resolve_statcodes <- function(in_ind, out_ind) {
+  dat <- readRDS(sc_retrieve(in_ind, remake_file = 'getters.yml')) %>%
+    ungroup() %>%
+  # drop values that are estimated or blank-corrected
+  # drop values that are not min, mean, max
+    filter(!ResultValueTypeName %in% c('Estimated', 'Blank Corrected Calc')) %>%
+    filter(StatisticalBaseCode %in% c(NA, 'Mean', 'Minimum', 'Maximum', 'mean',
+                                      'Geometric Mean', 'Daily Maximum', 'Daily Minimum',
+                                      'Daily Geometric Mean'))
+
+
+  # print message that says how many observations we lost when dropped
+  nrow_o <- nrow(readRDS(sc_retrieve(in_ind, remake_file = 'getters.yml')))
+  message(paste(nrow_o - nrow(dat), 'observations were dropped due to estimation, blank correction, or statcode that was not mean, min, max'))
```

Comment:
Curious what sort of numbers you're seeing here at the CONUS-extent for temperature data!